api/graphql/schema.graphql

```diff
@@ -608,7 +608,7 @@ type ApplicantFilter {
   expectedClassifications: [Classification]
     @belongsToMany(relation: "classifications")
   skills: [Skill] @belongsToMany
-  pools: [Pool] @belongsToMany @can(ability: "view", resolved: true)
+  pools: [Pool] @belongsToMany(scopes: ["authorizedToView"]) @can(ability: "view", resolved: true)
```

Comment:
The authorizedToView scope is unnecessary here, since all the pools attached to a filter should already by published.

Reply:
Or... Hmm. I'm assuming we should avoid using the scope if we can, since it makes the query more complex. If we decide to rely more heavily on this kind of scope, we might have to take a different stance. Another point for discussion in https://github.com/GCTC-NTGC/gc-digital-talent/discussions/6038 I suppose.

Reply:
Do we know if our seeders generate invalid data? 🤔 Removing the scopes just results in the same error we were already getting:

```js
[
    {
        "message": "This action is unauthorized.",
        "extensions": {
            "category": "authorization"
        },
        "path": [   "poolCandidateSearchRequests",  0,  "applicantFilter", "pools" ]
    }
]
```

Reply:
Okay, yeah. Seems like this could be an issue with seeders since we just attach a random pool in the factory.

https://github.com/GCTC-NTGC/gc-digital-talent/blob/f40006c6da0a723bd7d35c6968d2c22c2e2b5393/api/database/factories/ApplicantFilterFactory.php#L85-L88

Reply:
Updated seeders in [fix filter factories to only attach eligible pools](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6064/commits/fdee787f7911e28a51cb73398314ac55e07701a7) and removed scope in [remove unnecessary scope from applicant filter](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6064/commits/7576dfff980bff3d5f46615b3052a6f8beaf3f9f)